### PR TITLE
feat(linter): Split jest/prefer-todo into a shared rule and a vitest-specific rule.

### PR DESCRIPTION
**Pull request by connorshea**
Part of #21664.

Generated with Claude Code, Opus 4.7. Reviewed by me.

We should really improve the diagnostics for this rule, but that's a problem for later.

No changes to behavior or tests, other than the fix of a test that got mis-ported in the rule previously that Copilot pointed out.

### CODE REVIEW

File: crates/oxc_linter/src/rules/shared/jest_vitest/prefer_todo.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                } else {
+                    match &call_expr.callee {
+                        Expression::StaticMemberExpression(mem_expr) => {
+                            fixer.replace(mem_expr.property.span, "todo")
+                        }
+                        Expression::ComputedMemberExpression(mem_expr) => {
+                            fixer.replace(mem_expr.expression.span(), "'todo'")
+                        }
+                        _ => return fixer.delete_range(call_expr.span),
+                    }
```
[the remark this answers is not in the document]
Comment (connorshea) [reply crates/oxc_linter/src/rules/shared/jest_vitest/prefer_todo.rs:75]: this seems potentially correct, but I'd rather not change behavior here.

File: crates/oxc_linter/src/rules/jest/prefer_todo.rs [hunk 2]
```diff
@@ -190,7 +56,7 @@ fn tests() {
         ("it('foo', () => {})", None),
         ("test.skip('i need to write this test', () => {});", None),
         ("test.skip('i need to write this test', function() {});", None),
-        ("test[`skip`]('i need to write this test', function() {});", None),
+        (r#"test["skip"]('i need to write this test', function() {});"#, None),
```
Comment (camchenry) [inline crates/oxc_linter/src/rules/jest/prefer_todo.rs:59]: Is this intentional?

**Review by camchenry — APPROVED** [review]

File: crates/oxc_linter/src/rules/jest/prefer_todo.rs (hunk 2 above)
Comment (connorshea) [reply crates/oxc_linter/src/rules/jest/prefer_todo.rs:59]: Yeah, it was pointed out by Copilot so I updated this manually, to match the original implementation. https://github.com/jest-community/eslint-plugin-jest/blob/main/src/rules/__tests__/prefer-todo.test.ts#L67-L73

Comment (connorshea) [reply crates/oxc_linter/src/rules/jest/prefer_todo.rs:59]: It was a mistake in the rulegen porting I assume, we've had two identical tests on this rule since its creation ¯\_(ツ)_/¯

**Comment by connorshea** [comment]
### Merge activity

* **Apr 27, 4:14 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Apr 27, 4:14 AM UTC**: `connorshea` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
* **Apr 27, 4:20 AM UTC**: Merged by the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
